minecraft/src/main/java/net/fabricmc/loader/impl/game/minecraft/McVersion.java

```diff
@@ -32,12 +42,22 @@
 	private final String normalized;
 	private final OptionalInt classVersion;
 
-	private McVersion(String name, String release, OptionalInt classVersion) {
-		this.raw = name;
-		this.normalized = McVersionLookup.normalizeVersion(name, release);
+	private McVersion(String id, String name, String raw, String release, OptionalInt classVersion) {
+		this.id = id;
+		this.name = name;
+		this.raw = raw;
+		this.normalized = McVersionLookup.normalizeVersion(raw, release);
 		this.classVersion = classVersion;
 	}
 
+	public String getId() {
+		return id;
+	}
+
+	public String getName() {
```

Comment:
Why not record?

Reply:
Fabric Loader is 100% Java 8

Reply:
> Fabric Loader is 100% Java 8

for compatibility with older versions of MC?

Reply:
yes, don't want to introduce incompatibility with old versions or old hardware (e.g. old intel on windows) for such little gain